crates/types/src/config/log_server.rs

```diff
@@ -86,6 +93,20 @@ impl LogServerOptions {
         self.rocksdb_disable_wal_fsync
     }
 
+    pub fn rocksdb_max_sub_compactions(&self) -> u32 {
+        if self.rocksdb_max_sub_compactions == 0 {
+            let parallelism: NonZeroU32 = std::thread::available_parallelism()
```

Comment:
Is it a problem that `available_parallelism` might overcount when running in a VM with CPU usage limits (according to the docs)? I mean we are using this function at other places as well so it's not unique to this PR.

Reply:
This can be a problem although it should be rare. The standard practice is to use `num_cpus`. This is something we can do in a follow up.